CRM/Contribute/Page/ContributionRecurTab.php

```diff
@@ -0,0 +1,493 @@
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2015
+ * $Id$
+ *
+ */
+class CRM_Contribute_Page_ContributionRecurTab extends CRM_Core_Page {
+
+  /**
+   * The action links that we need to display for the browse screen.
+   *
+   * @var array
+   */
+  static $_links = NULL;
+  public $_permission = NULL;
+  public $_contactId = NULL;
+  public $_crid = NULL;
+
+  /**
+   * This method returns the links that are given for recur search row.
+   * currently the links added for each row are:
+   * - View
+   * - Edit
+   * - Cancel
+   *
+   * @param bool $recurID
+   * @param string $context
+   *
+   * @return array
+   */
+  public static function &recurLinks($recurID = FALSE, $context = 'contribution') {
+    if (!(self::$_links)) {
+      self::$_links = array(
+        CRM_Core_Action::VIEW => array(
+          'name' => ts('View'),
+          'title' => ts('View Recurring Payment'),
+          'url' => 'civicrm/contact/view/contributionrecur',
+          'qs' => "reset=1&id=%%crid%%&cid=%%cid%%&context={$context}",
+        ),
+        CRM_Core_Action::UPDATE => array(
+          'name' => ts('Edit'),
+          'title' => ts('Edit Recurring Payment'),
+          'url' => 'civicrm/contributionrecur/add',
+          'qs' => "reset=1&action=update&crid=%%crid%%&cid=%%cid%%&context={$context}",
+        ),
+        CRM_Core_Action::DELETE => array(
+          'name' => ts('Delete'),
+          'title' => ts('Delete Recurring Payment'),
+          'url' => 'civicrm/contributionrecur/add',
+          'qs' => "reset=1&action=delete&crid=%%crid%%&cid=%%cid%%&context={$context}",
+        ),
+        CRM_Core_Action::DISABLE => array(
+          'name' => ts('Cancel'),
+          'title' => ts('Cancel'),
+          'ref' => 'crm-enable-disable',
+        ),
+      );
+    }
+    
+    if ($recurID) {
+      $paymentProcessorObj = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($recurID, 'recur', 'obj');
+      if (is_object($paymentProcessorObj) && $paymentProcessorObj->isSupported('cancelSubscription')) {
```

Comment:
preferred syntax is $paymentProcessorObj->supports('cancelSubscription');

This does mean that there needs to be a new method on CRM_Core_Payment - ie. supportsCancelSubscription

I hadn't spotted that there was this alternate method in use until now :-( 

The nice thing about the supportsCancelSubscription approach is it tends to create a logical place for a comment block giving some detail. It does mean 2 functions rather than one to indicate support - but most of that is just comment.